core.ts

```diff
@@ -200,845 +200,844 @@ class Poll<
 	}
 }
 
-export function PinejsClientCoreFactory(
-	Promise: PinejsClientCoreFactory.PromiseRejector,
-): typeof PinejsClientCoreFactory.PinejsClientCore {
-	if (!isPromiseRejector(Promise)) {
-		throw new Error('The Promise implementation must support .reject');
-	}
+const transformGetResult = (params: PinejsClientCoreFactory.Params) => {
+	const singular = isObject(params) && params.id != null;
 
-	const isPrimitive = (
-		value?: any,
-	): value is null | string | number | boolean | Date => {
-		return (
-			value === null ||
-			isString(value) ||
-			NumberIsFinite(value) ||
-			isBoolean(value) ||
-			isDate(value)
-		);
-	};
-
-	const baseEscapeResource = (resource: string | string[]) => {
-		if (isString(resource)) {
-			return encodeURIComponent(resource);
-		} else if (Array.isArray(resource)) {
-			return resource.map(encodeURIComponent).join('/');
-		} else {
-			throw new Error('Not a valid resource: ' + typeof resource);
+	return (data: { d: any[] }): PinejsClientCoreFactory.PromiseResultTypes => {
```

Comment:
Can we type this any more strongly?

Reply:
Not really, it'd need to be part of a whole project to have types based on what the query is and therefore what the api response should be